cookbook/arduino_port_extender.rst

```diff
@@ -0,0 +1,183 @@
+Arduino Port Extender
+=====================
+
+.. seo::
+    :description: Instructions on using an Arduino board, like the Pro Mini for extending ports of a ESPHome node
+    :image: arduino_pro_mini.jpg
+    :keywords: Arduino port extender ESPHome
+
+The Arduino is a widely used microcontroller, easy to use and accesible.
+With this sketch you can use any compatible Arduino board with ESPHome and read and write pins.
+
+
+.. figure:: images/arduino_pro_mini.jpg
+    :align: center
+    :width: 75.0%
+
+
+The Arduino is connected to the ESP via I²C, common Arduinos uses pins ``A4`` and ``A5`` for this
+so those pins are not available to read from ESPHome.
+It is recommended to use a 3.3V Arduino, however using 5V arduinos seems to work too. In the last
+case you should power your 5V Arduino with 3.3V otherwise you will need a level converter for the
```

Comment:
I understand your concern, this is something you should not recommend to do, however I explain it seems to work because I believe many people will fall in this case, having and ESP and a 5V Arduino.
I have 10 boards which were using a 5v arduino pro mini, powered with 12V and regulated down to 5 using a 7805 IC. They were using the NRF24 radio, and I decided to convert them to WiFi and long story short ended up here.
I changed the 7805 by the LD1117V33 which can easily do 950mA, and had no issue with any of that. 
I did some investigation just in case, found out the chips are the same but 3.3 are run at 8Mhz instead of 16Mhz.
In the desktop breadboard I have a LoLin ESP8266, I've powered two 5V pro minis and connected via I2C no problem either (with the LoLin 3v3 regulator.